jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/StepsRunner.java

```diff
@@ -566,11 +568,35 @@ private void pushImages() {
         });
   }
 
+  private void pushManifestList() {
+    ProgressEventDispatcher.Factory childProgressDispatcherFactory =
+        Verify.verifyNotNull(rootProgressDispatcher).newChildProducer();
+
+    results.buildResult =
+        executorService.submit(
+            () -> {
+              realizeFutures(results.singleManifestPushResults.get());
+              List<Future<BuildResult>> manifestListPushResults =
+                  scheduleCallables(
+                      PushImageStep.makeListPushManifestList(
+                          buildContext,
+                          childProgressDispatcherFactory,
```

Comment:
Hmm... I saw some test failures. I suspect there's a problem with using a progress dispatcher here.

Reply:
Yes there was a  problem with the dispatcher but its all sorted out now.

Reply:
Hmm... I don't really get how the last commit fixed the test failure. It shouldn't affect the progress completion ratio.

Reply:
````  private void pushManifestList() {
    ProgressEventDispatcher.Factory childProgressDispatcherFactory =
        Verify.verifyNotNull(rootProgressDispatcher).newChildProducer();

    results.buildResults =
        executorService.submit(
            () -> {
              realizeFutures(results.pushImageResults.get());
              if (results.builtImagesAndBaseImages.get().size() == 1) {
                return results.pushImageResults.get();
              }

              return scheduleCallables(
                  PushImageStep.makeListPushManifestList(
                      buildContext,
                      childProgressDispatcherFactory,
                      results.targetRegistryClient.get(),
                      results.manifestListOrSingleManifest.get(),
                      results.manifestCheckResult.get().isPresent()));
            });
```

Reply:
Ah, I see the last test failure was not because of the progress but a digest mistach.

In light of this, probably the short-circuiting in https://github.com/GoogleContainerTools/jib/pull/2724#discussion_r476679646 should go inside the try-resource block.